docker-compose.yml

```diff
@@ -2,26 +2,30 @@ notaryserver:
   build: .
   dockerfile: server.Dockerfile
   links:
-   - notarymysql
-   - notarysigner
-  ports:
-   - "8080"
-   - "4443:4443"
+    - notarymysql
+    - notarysigner
   environment:
-   - SERVICE_NAME=notary
-  command: -config=fixtures/server-config.json
+    - SERVICE_NAME="notaryserver"
+  ports:
+    - "8080"
+    - "4443:4443"
+  entrypoint: /bin/bash
+  command: -c "./migrations/migrate.sh && notary-server -config=fixtures/server-config.json"
 notarysigner:
-  volumes:
-   - /dev/bus/usb/003/010:/dev/bus/usb/002/010
-   - /var/run/pcscd/pcscd.comm:/var/run/pcscd/pcscd.comm
   build: .
   dockerfile: signer.Dockerfile
   links:
-   - notarymysql
-  command: -config=fixtures/signer-config.json
+    - notarymysql
+  environment:
+    - SERVICE_NAME="notarysigner"
+  entrypoint: /bin/bash
+  command: -c "./migrations/migrate.sh && notary-signer -config=fixtures/signer-config.json"
 notarymysql:
   volumes:
-    - notarymysql:/var/lib/mysql
-  build: ./notarymysql/
+    - ./notarymysql/docker-entrypoint-initdb.d:/docker-entrypoint-initdb.d
+    - ./notarymysql/data:/var/lib/mysql
```

Comment:
```
docker-compose rm && docker rmi notary_notarymysql && docker volume rm notarymysql
docker-compose build && docker-compose up
``` 
- With local volume only (no named-volume): --> permissions errors
- With named volume for data --> similarly to @diogomonica, migrations didn't seem to run:

```
Database initialized
notarymysql_1  | MySQL init process in progress...
...
notarymysql_1  | 2016-02-05 22:57:44 139955359193024 [Note] mysqld: ready for connections.
notarymysql_1  | Version: '10.1.10-MariaDB-1~jessie'  socket: '/var/run/mysqld/mysqld.sock'  port: 0  mariadb.org binary distribution
notarymysql_1  | Warning: Unable to load '/usr/share/zoneinfo/leap-seconds.list' as time zone. Skipping it.
notarymysql_1  | 2016-02-05 22:57:45 139955358635776 [Warning] 'proxies_priv' entry '@% root@9af5dba6531b' ignored in --skip-name-resolve mode.
notarymysql_1  | 
notarymysql_1  | /docker-entrypoint.sh: running /docker-entrypoint-initdb.d/initial-notaryserver.sql
notarymysql_1  | 
notarymysql_1  | 
notarymysql_1  | /docker-entrypoint.sh: running /docker-entrypoint-initdb.d/initial-notarysigner.sql
notarymysql_1  | 
notarymysql_1  | 
notarymysql_1  | 2016-02-05 22:57:45 139955358336768 [Note] mysqld: Normal shutdown
notarymysql_1  | 
notarymysql_1  | 2016-02-05 22:57:45 139955358336768 [Note] Event Scheduler: Purging the queue. 0 events
notarymysql_1  | 2016-02-05 22:57:45 139954558560000 [Note] InnoDB: FTS optimize thread exiting.
notarymysql_1  | 2016-02-05 22:57:45 139955358336768 [Note] InnoDB: Starting shutdown...
notarymysql_1  | 2016-02-05 22:57:47 139955358336768 [Note] InnoDB: Shutdown completed; log sequence number 1616829
notarymysql_1  | 2016-02-05 22:57:47 139955358336768 [Note] mysqld: Shutdown complete
notarymysql_1  | 
notarymysql_1  | 
notarymysql_1  | MySQL init process done. Ready for start up.
notarymysql_1  | 
notarymysql_1  | 2016-02-05 22:57:47 140308565714880 [Note] mysqld (mysqld 10.1.10-MariaDB-1~jessie) starting as process 1 ...
...
notaryserver_1 | {"http.request.host":"notary-server:4443","http.request.id":"2fcd1417-2326-4ea1-8243-adfd4ff43ca8","http.request.method":"GET","http.request.remoteaddr":"192.168.99.1:57669","http.request.uri":"/v2/","http.request.useragent":"Go-http-client/1.1","http.response.duration":"39.871µs","http.response.status":200,"http.response.written":2,"level":"info","msg":"response completed","time":"2016-02-05T22:58:19Z"}
notaryserver_1 | {"level":"debug","msg":"retrieving timestamp key for docker.io/cyli/5:timestamp","time":"2016-02-05T22:58:19Z"}
notaryserver_1 | {"docker.io/cyli/5":"gun","http.request.host":"notary-server:4443","http.request.id":"8edaa932-b924-49fa-90da-6012bde45f09","http.request.method":"GET","http.request.remoteaddr":"192.168.99.1:57670","http.request.uri":"/v2/docker.io/cyli/5/_trust/tuf/timestamp.key","http.request.useragent":"Go-http-client/1.1","level":"error","msg":"500 GET timestamp key: Error 1146: Table 'notaryserver.timestamp_keys' doesn't exist","time":"2016-02-05T22:58:19Z"}
notaryserver_1 | {"http.request.host":"notary-server:4443","http.request.id":"8edaa932-b924-49fa-90da-6012bde45f09","http.request.method":"GET","http.request.remoteaddr":"192.168.99.1:57670","http.request.uri":"/v2/docker.io/cyli/5/_trust/tuf/timestamp.key","http.request.useragent":"Go-http-client/1.1","http.response.contenttype":"application/json; charset=utf-8","http.response.duration":"3.402733ms","http.response.status":500,"http.response.written":145,"level":"info","msg":"response completed","time":"2016-02-05T22:58:19Z"}
notaryserver_1 | 
notaryserver_1 | (Error 1146: Table 'notaryserver.timestamp_keys' doesn't exist)
```